fluss-common/src/main/java/com/alibaba/fluss/metadata/TablePath.java

```diff
@@ -179,7 +179,9 @@ private static boolean containsInvalidPattern(String identifier) {
                             || (c >= '0' && c <= '9')
                             || (c >= 'A' && c <= 'Z')
                             || c == '_'
-                            || c == '-';
+                            || c == '-'
+                            || c == '$';
```

Comment:
After reconsidering, we cannot allow the `$` character in table names because we use `$lake` as a suffix for lake tables in the Flink catalog. If a table name contains `$lake`, the catalog would be unable to distinguish whether it refers to a regular table or a lake table, leading to potential ambiguity.

We can revert the changes on `detectInvalidName`, but split the `partitionName` using `$`, and validate each partition value using `detectInvalidName`.

Reply:
Done!